components/ILIAS/Authentication/classes/LoginPage/LoginPageLanguagesOverviewTable.php

```diff
@@ -0,0 +1,226 @@
+     */
+    private ?array $records = null;
+
+    public function __construct(
+        private readonly ilCtrl $ctrl,
+        private readonly ilLanguage $lng,
+        \ILIAS\HTTP\Services $http,
+        private readonly \ILIAS\UI\Factory $ui_factory,
+        private readonly \ILIAS\UI\Renderer $ui_renderer
+    ) {
+        $this->request = $http->request();
+        $this->data_factory = new Data\Factory();
+    }
+
+    public function getComponent(): UI\Component\Table\Data
+    {
+        $columns = $this->getColumns();
+        $actions = $this->getActions();
+
+        return $this->ui_factory
+            ->table()
+            ->data($this->lng->txt('login_pages'), $columns, $this)
+            ->withActions($actions)
+            ->withRequest($this->request);
+    }
+
+    /**
+     * @return array<string, \ILIAS\UI\Component\Table\Column\Column>
+     */
+    private function getColumns(): array
+    {
+        return [
+            'language' => $this->ui_factory
+                ->table()
+                ->column()
+                ->text($this->lng->txt('login_page'))
+                ->withIsSortable(false),
+            'status_icon' => $this->ui_factory
+                ->table()
+                ->column()
+                ->text($this->lng->txt('active'))
+                ->withIsSortable(false)
+        ];
+    }
+
+    /**
+     * @return array<string, \ILIAS\UI\Component\Table\Action\Action>
+     */
+    protected function getActions(): array
+    {
+        $query_params_namespace = ['loginpage', 'languages'];
+
+        $overview_uri = $this->data_factory->uri(
+            ILIAS_HTTP_PATH . '/' . $this->ctrl->getLinkTargetByClass(
+                \ilAuthLoginPageEditorGUI::class,
+                'handleLoginPageActions'
+            )
+        );
+
+        $overview_url_builder = new UI\URLBuilder($overview_uri);
+        [
+            $overview_url_builder,
+            $overview_action_parameter,
+            $overview_row_id
+        ] = $overview_url_builder->acquireParameters(
+            $query_params_namespace,
+            'action',
+            'key'
+        );
+
+        return [
+            'edit' => $this->ui_factory->table()->action()->single(
+                $this->lng->txt('edit'),
+                $overview_url_builder->withParameter($overview_action_parameter, 'edit'),
+                $overview_row_id
+            ),
+            'activate' => $this->ui_factory->table()->action()->standard(
+                $this->lng->txt('login_page_activate'),
+                $overview_url_builder->withParameter($overview_action_parameter, 'activate'),
+                $overview_row_id
+            ),
+            'deactivate' => $this->ui_factory->table()->action()->standard(
+                $this->lng->txt('login_page_deactivate'),
+                $overview_url_builder->withParameter($overview_action_parameter, 'deactivate'),
+                $overview_row_id
+            )
```

Comment:
Please move `activate` and `deactivate` to a private class constant.

Reply:
ok. What about 'edit' in line 110?

Reply:
Even if it is used only once, you SHOULD (but not MUST IMO) add it.

Reply:
ok. I add it